sso-kit-client/sso-kit-client-lit/src/SsoKit.ts

```diff
@@ -0,0 +1,168 @@
+     */
+    authenticated = false;
+
+    /**
+     * The user roles
+     */
+    roles: string[] = [];
+
+    /**
+     * The URL which will be called to log in to the SSO provider
+     */
+    loginUrl: string | undefined = undefined;
+
+    /**
+     * The URL which will be called to log out from the SSO provider
+     */
+    logoutUrl: string | undefined = undefined;
+
+    /**
+     * If true, the app will listen to back-channel logout events
+     */
+    backChannelLogoutEnabled = false;
+
+    /**
+     * If true, the user has been logged out from the SSO provider
+     */
+    backChannelLogoutHappened = false;
+
+    /**
+     * A list of authentication providers. The first provider is the main provider.
+     */
+    registeredProviders: string[] = [];
+
+    /**
+     * The subscription to the back-channel logout event
+     */
+    private logoutSubscription: Subscription<any> | undefined;
+
+    constructor() {
+        makeAutoObservable(this);
+
+        const authInfo = JSON.parse(Buffer.from(Hilla.SSO!, 'base64').toString('utf-8')) as SingleSignOnData;
+        this.authenticated = authInfo.authenticated;
+        this.roles = authInfo.roles;
+        this.loginUrl = authInfo.loginLink;
+        this.logoutUrl = authInfo.logoutLink;
+        this.backChannelLogoutEnabled = authInfo.backChannelLogoutEnabled;
+
+        // @ts-ignore: the imported file might not exist, but in that case registeredProviders will be empty
+        import("Frontend/generated/SingleSignOnEndpoint").then((endpoint) => {
+            this.registeredProviders = endpoint.getRegisteredProviders();
+        });
+
+        if (this.authenticated && this.backChannelLogoutEnabled) {
+            // @ts-ignore: the imported file might not exist, but in that case backChannelLogoutEnabled will be false
+            import("Frontend/generated/BackChannelLogoutEndpoint").then((endpoint) => {
+                this.logoutSubscription = endpoint.subscribe();
+                this.logoutSubscription!.onNext(() => {
+                    this.backChannelLogoutHappened = true;
+                    this.logoutSubscription!.cancel();
+                });
+            });
+        }
+    }
+
+    /**
+     * Clears the authentication information
+     */
+    clearAuthInfo = () => {
+        this.authenticated = false;
+        this.roles = [];
+        this.logoutUrl = undefined;
+        this.backChannelLogoutHappened = false;
+        if (this.logoutSubscription) {
+            this.logoutSubscription = undefined;
+        }
+    }
+
+    /**
+     * Returns the authentication providers' login URLs.
+     */
+    get loginUrls() {
+        // create a map of providers as keys and login URLs as values
+        return this.registeredProviders.map((provider) => ({ name: provider, link: loginUrl(provider) }));
+    }
+
+    /**
+     * Returns true if the user is logged in
+     */
+    get loggedIn() {
+        return this.authenticated;
```

Comment:
If `loggedIn` and `authenticated` mean the same thing, let's use the same term.

Reply:
Removed this function. The `authenticated` variable itself is also exported, so it can be used to check if the user is uthenticated.